### ISSUE TITLE
Function returning spread operator infers wrong type and get properties can be accessed

### ISSUE THREAD

**Issue by michaelcheers** [microsoft/TypeScript#44543]
# Bug Report

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?target=99&jsx=0&ts=4.4.0-dev.20210610#code/MYGwhgzhAEAqCmEAuBvA5vJ0Ay8B2aSAFgBQCUKATpgK6V7QCMA3AL6sBQHAZjXsEgCWAewYQADtTAATADywAfNBJIAXHDLQU0akjoMUAOmPQkraJxCZTiJI2gBeLccN54Adzi3yrZh2CiEMJWhiDCaCq2jIa4BMRkzNAA9EnQAKIAHuLwAvDS6sSUwu4w8JRFlIbQAIICNGAgBURFJdBlFYYcVlhItgBMjsrCjkokRsbCrGRkJG6eCMjkCf6BwfCh4ZHIfTH4hEQJyamZ2bn5ps3FpeXClTV1DU0t1x1d1r3IAMyDElLSsx4vItpn4AnggiEwhEPkhPrs4gdEil0lkcr1zoUrm0bndanpHtBpMJEHgAOQ9S6edq3QxAA)

### 💻 Code

```ts
class Test{get Length(){return 1;}}

function spread<T> (t: T) { return {... t} }
let test1 = {...new Test()};
console.log(test1.Length); // Expected: throws error. Actual: throws error.
let test2 = (o => ({...o}))(new Test());
console.log(test2.Length); // Expected: throws error. Actual: throws error.
let test3 = spread(new Test());
console.log(test3.Length); // Expected: throws error. Actual: doesn't throw error.
```

### 🙁 Actual behavior

Doesn't throw compile-time error.
Error is at runtime, the spread operator of Test does not copy over properties.

### 🙂 Expected behavior

Typescript's type system should figure out that the type is not T but instead is just enumerable properties of T.

**Comment by RyanCavanaugh** [comment]
See #9726
